src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1780,6 +1783,31 @@ private void VisitSearch(MethodCallExpression searchExpression)
 
             if (((SearchOptions)value).HasFlag(SearchOptions.Guess))
                 _chainedWhere = true;
+            
+            return;
+
+            void WhereExistsAndNegatedSearch(ExpressionInfo expressionInfo, string searchTerms, SearchOperator @operator)
+            {
+                DocumentQuery.OpenSubclause();
+                DocumentQuery.WhereExists(expressionInfo.Path);
+                DocumentQuery.AndAlso();
+                DocumentQuery.NegateNext();
+                DocumentQuery.Search(expressionInfo.Path, searchTerms, @operator);
+                DocumentQuery.CloseSubclause();
+            }
+            
+            bool IsPreviousSearchOnSameField(Expression searchTargetExpression, Expression currentSearchExpression)
+            {
+                if (searchTargetExpression is MethodCallExpression targetMce && currentSearchExpression is MethodCallExpression currentSearchMce)
+                {
+                    var targetExpressionInfo = GetMember(targetMce.Arguments[1]);
+                    var currentSearchExpressionInfo = GetMember(currentSearchMce.Arguments[1]);
```

Comment:
do we have a guarantee that number of arguments is at least 2 here? always?

Reply:
From what I've checked these arguments are corresponding to `Search` method arguments from `LinqExtensions`